src/main/java/com/Elessar/app/server/RegisterHandler.java

```diff
@@ -0,0 +1,56 @@
+package com.Elessar.app.server;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.Elessar.proto.Registration.RegistrationResponse;
+import com.Elessar.proto.Registration.RegistrationRequest;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * Created by Hans on 1/15/19.
+ */
+public class RegisterHandler implements HttpHandler {
+    private final Map<String, RegistrationRequest> userData;
+
+    public RegisterHandler(Map<String, RegistrationRequest> userData) {
+        this.userData = userData;
+    }
+
+    @Override
+    public void handle(HttpExchange he) throws IOException {
+        final String requestType = he.getRequestMethod();
+        // Only handle POST request
+        if (!"POST".equals(requestType)) {
+            final String response = "NOT IMPLEMENTED\n";
+            he.sendResponseHeaders(501, response.length());  // 501 tells the caller that this method is not supported by the server
+            try (OutputStream os = he.getResponseBody()) {
+                os.write(response.getBytes());
+            }
+            return ;
+        }
+
+        try (InputStream is = he.getRequestBody()) {
+            RegistrationRequest regRequest = RegistrationRequest.parseFrom(is);
+            if (userData.containsKey(regRequest.getName())) {
+                final RegistrationResponse.Builder regResponse = RegistrationResponse.newBuilder();
```

Comment:
A side note just in case you did not know:
Those "setXXX()" functions from builder class returns the builder object so you can write it like this:
final RegistrationResponse response = RegistrationResponse.newBuilder().setSuccess(false).setFailReason(...).build();